src/Apps/IdentityVerification/IdentityVerificationApp.tsx

```diff
@@ -1,35 +1,131 @@
+// import { RedirectException } from "found"
+
+const logger = createLogger("IdentityVerificationApp.tsx")
 
 interface Props {
   me: IdentityVerificationApp_me
+  relay: RelayProp
+  // match: any
 }
 
-const IdentityVerificationApp: React.FC<Props> = ({ me }) => {
+const IdentityVerificationApp: React.FC<Props> = ({ me, relay }) => {
+  // if (!me) {
+  //   const loginPath =
+  //     "/log_in?redirect_uri=" + encodeURIComponent(match.location.pathname)
+  //   throw new RedirectException(loginPath)
+  // }
   const { identityVerification } = me
+  const { environment } = relay
+
+  const [requesting, setRequesting] = useState(false)
+  const [showErrorModal, setShowErrorModal] = useState(false)
 
   if (!identityVerification || identityVerification.userID !== me.internalID) {
     return <ErrorPage code={404} />
   }
 
   const { trackEvent } = useTracking()
 
-  const clickContinueToVerification = () => {
+  const trackClickedContinueToVerification = () => {
     trackEvent({
       context_page_owner_id: identityVerification.internalID,
       action_type: Schema.ActionType.ClickedContinueToIdVerification,
       context_page: Schema.PageName.IdentityVerificationPage,
     })
   }
 
+  function startIdentityVerification() {
+    const mutation = new Promise<
+      IdentityVerificationAppStartFlowMutationResponse
+    >((resolve, reject) => {
+      commitMutation<IdentityVerificationAppStartFlowMutation>(environment, {
+        onCompleted: data => resolve(data),
+        onError: error => reject({ error }),
+        mutation: graphql`
+          mutation IdentityVerificationAppStartFlowMutation(
+            $input: startIdentityVerificationMutationInput!
+          ) {
+            startIdentityVerification(input: $input) {
+              startIdentityVerificationResponseOrError {
+                ... on StartIdentityVerificationSuccess {
+                  identityVerificationFlowUrl
+                }
+                ... on StartIdentityVerificationFailure {
+                  mutationError {
+                    detail
+                    error
+                    message
+                  }
+                }
+              }
+            }
+          }
+        `,
+        variables: {
+          input: { identityVerificationId: identityVerification.internalID },
+          // input: { identityVerificationId: "fffffff" }, // for testing an error
+        },
+      })
+    })
+
+    mutation
+      .then(handleMutationSuccess)
+      .catch(handleMutationError)
+      .finally(() => {
+        setRequesting(false)
+      })
+  }
+
+  const handleMutationSuccess = (
+    result: IdentityVerificationAppStartFlowMutationResponse
+  ) => {
+    const {
+      startIdentityVerification: {
+        startIdentityVerificationResponseOrError: {
+          identityVerificationFlowUrl,
+          mutationError,
+        },
+      },
+    } = result
+    if (mutationError) {
+      logger.error("FAIL in handleMutation")
```

Comment:
Unless handled elsewhere, maybe log the mutation error from the logger?

Reply:
yes, this was just debugging/understanding why the `onError` code wasn't being executed (because that is only if the xhr request itself fails, need to check for error in `onCompleted`). If there is a _better_ way of doing this than what i am doing (where onError basically takes 2 different signatures) I'm interested to hear it!